### DISCUSSION TITLE
Weird results in join

### DISCUSSION THREAD

**Discussion by ghost** [jOOQ/jOOQ#15913] [Q&A] [answered]
Hi

I have this query with the usual bank/accounts/transactions database:

```
            final var accounts = table("accounts").as("a");
            final var transactions = table("transactions").as("t");
            final var query = context
                    .select(field("a.id"), field("t.*"))
                    .from(accounts)
                    .join(transactions)
                    .on(field("a.id").eq(field("t.account_id")))
                    .fetch();
```

From my understanding this would be a LEFT OUTER JOIN? Executing this query directly in pgAdmin returns all columns for the transactions table. jOOQ only returns account id + transaction ids from what I saw, the other columns from the transaction table are missing? Could you tell me where/what could be wrong?

Thanks!

**Comment by lukaseder** [comment, marked as answer]
A `Field` is a single `Field`. If you put an asterisk in a plain SQL Field, jOOQ will still think there's only one field, and it won't be able to correctly fetch all the remaining columns. Use an explicit asterisk instead:

https://www.jooq.org/doc/latest/manual/sql-building/sql-statements/select-statement/select-clause/select-clause-asterisk

Probably a typo, but just to be sure, to run a left join, use `leftJoin()`, not `join()`
